redis/src/main/scala/scalacache/redis/ShardedRedisCache.scala

```diff
@@ -0,0 +1,131 @@
+package scalacache.redis
+
+import java.nio.charset.Charset
+
+import com.typesafe.scalalogging.StrictLogging
+import redis.clients.jedis._
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import scalacache.{ Cache, LoggingSupport }
+
+class ShardedRedisCache(jedisPool: ShardedJedisPool, override val customClassloader: Option[ClassLoader] = None)(implicit execContext: ExecutionContext = ExecutionContext.global)
+    extends Cache
+    with RedisSerialization
+    with LoggingSupport
+    with StrictLogging {
+
+  import ShardedRedisCache.StringWithUtf8Bytes
+
+  /**
+   * Get the value corresponding to the given key from the cache
+   * @param key cache key
+   * @tparam V the type of the corresponding value
+   * @return the value, if there is one
+   */
+  override def get[V](key: String) = Future {
```

Comment:
All the methods apart from `removeAll` are straight copy-pastes of the ones in `RedisCache`, and some of them are quite complex, so I'd like to share the implementations if possible. Could you refactor your code to reduce the code duplication?

Perhaps you could introduce a common trait that both `RedisCache` and `ShardedRedisCache` extend. Both `Jedis` and `ShardedJedis` implement `JedisCommands`, so you should be able to change the `withJedisClient` method to take a `f: JedisCommands => T`. (You'll also need a separate `withShardedJedisClient` method in `ShardedRedisCache` to use for sharded-specific methods like `removeAll`.)